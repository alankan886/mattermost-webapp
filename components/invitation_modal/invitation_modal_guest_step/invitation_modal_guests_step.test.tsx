// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
// See LICENSE.txt for license information.

import React from 'react';
import {shallow} from 'enzyme';

import InvitationModalGuestsStep from './index';

describe('components/invitation_modal/InvitationModalGuestsStep', () => {
    const props = {
        teamName: 'Test Team',
        currentTeamId: 'test-team-id',
        inviteId: '123',
        searchProfiles: jest.fn(),
        emailInvitationsEnabled: true,
        onSubmit: jest.fn(),
        onEdit: jest.fn(),
        currentUsers: 4,
        myInvitableChannels: [],
        searchChannels: jest.fn(),
    };
    test('should match the snapshot', () => {
        const wrapper = shallow(
            <InvitationModalGuestsStep
                {...props}
            />,
        );
        expect(wrapper).toMatchSnapshot();
    });

    test('should match the snapshot when email invitations are disabled', () => {
        const wrapper = shallow(
            <InvitationModalGuestsStep
                {...props}
            />,
        );
        expect(wrapper).toMatchSnapshot();
    });
});
