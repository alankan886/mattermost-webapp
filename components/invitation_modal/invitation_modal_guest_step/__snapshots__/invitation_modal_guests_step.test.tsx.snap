// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/invitation_modal/InvitationModalGuestsStep should match the snapshot 1`] = `
<div
  className="InvitationModalGuestsStep"
>
  <div
    className="modal-icon"
  >
    <InviteIcon />
  </div>
  <h1
    id="invitation_modal_title"
  >
    <injectIntl(FormattedMarkdownMessage)
      defaultMessage="Invite **Guests** to {teamName}"
      id="invitation_modal.guests.title"
      values={
        Object {
          "teamName": "Test Team",
        }
      }
    />
  </h1>
  <div
    className="add-people"
    data-testid="addPeople"
  >
    <h5>
      <MemoizedFormattedMessage
        defaultMessage="Invite People"
        id="invitation_modal.guests.add_people.title"
      />
    </h5>
    <div
      data-testid="emailPlaceholder"
    >
      <UsersEmailsInput
        ariaLabel="Invite People"
        emailInvitationsEnabled={true}
        inputValue=""
        loadingMessageDefault="Loading"
        loadingMessageId="widgets.users_emails_input.loading"
        noMatchMessageDefault="No one found matching **{text}**, type email to invite"
        noMatchMessageId="invitation_modal.guests.users_emails_input.no_user_found_matching"
        onChange={[Function]}
        onInputChange={[Function]}
        placeholder="Add guests or email addresses"
        showError={false}
        usersLoader={[Function]}
        validAddressMessageDefault="Invite **{email}** as a guest"
        validAddressMessageId="invitation_modal.guests.users_emails_input.valid_email"
        value={Array []}
      />
    </div>
    <div
      className="help-text"
    >
      <MemoizedFormattedMessage
        defaultMessage="Add existing guests or send email invites to new guests."
        id="invitation_modal.guests.add_people.description"
      />
    </div>
  </div>
  <div
    className="add-channels"
    data-testid="channelPlaceholder"
  >
    <h5>
      <MemoizedFormattedMessage
        defaultMessage="Search and Add Channels"
        id="invitation_modal.guests.add_channels.title"
      />
    </h5>
    <div>
      <MemoizedFormattedMessage
        defaultMessage="Search and add channels"
        id="invitation_modal.guests.add_channels.placeholder"
      >
        <Component />
      </MemoizedFormattedMessage>
    </div>
    <div
      className="help-text"
    >
      <MemoizedFormattedMessage
        defaultMessage="Specify the channels the guests have access to."
        id="invitation_modal.guests.add-channels.description"
      />
    </div>
  </div>
  <div
    className="custom-message"
    data-testid="customMessage"
  >
    <a
      href="#"
      onClick={[Function]}
    >
      <MemoizedFormattedMessage
        defaultMessage="Set a custom message"
        id="invitation_modal.guests.custom-message.link"
      />
    </a>
    <div
      className="help-text"
    >
      <MemoizedFormattedMessage
        defaultMessage="Create a custom message to make your invite more personal."
        id="invitation_modal.guests.custom-message.description"
      />
    </div>
  </div>
  <div
    className="invite-guests"
  >
    <button
      className="btn btn-inactive"
      disabled={true}
      id="inviteGuestButton"
      onClick={[Function]}
    >
      <MemoizedFormattedMessage
        defaultMessage="Invite Guests"
        id="invitation_modal.guests.invite_button"
      />
    </button>
  </div>
</div>
`;

exports[`components/invitation_modal/InvitationModalGuestsStep should match the snapshot when email invitations are disabled 1`] = `
<div
  className="InvitationModalGuestsStep"
>
  <div
    className="modal-icon"
  >
    <InviteIcon />
  </div>
  <h1
    id="invitation_modal_title"
  >
    <injectIntl(FormattedMarkdownMessage)
      defaultMessage="Invite **Guests** to {teamName}"
      id="invitation_modal.guests.title"
      values={
        Object {
          "teamName": "Test Team",
        }
      }
    />
  </h1>
  <div
    className="add-people"
    data-testid="addPeople"
  >
    <h5>
      <MemoizedFormattedMessage
        defaultMessage="Invite People"
        id="invitation_modal.guests.add_people.title"
      />
    </h5>
    <div
      data-testid="emailPlaceholder"
    >
      <UsersEmailsInput
        ariaLabel="Invite People"
        emailInvitationsEnabled={true}
        inputValue=""
        loadingMessageDefault="Loading"
        loadingMessageId="widgets.users_emails_input.loading"
        noMatchMessageDefault="No one found matching **{text}**, type email to invite"
        noMatchMessageId="invitation_modal.guests.users_emails_input.no_user_found_matching"
        onChange={[Function]}
        onInputChange={[Function]}
        placeholder="Add guests or email addresses"
        showError={false}
        usersLoader={[Function]}
        validAddressMessageDefault="Invite **{email}** as a guest"
        validAddressMessageId="invitation_modal.guests.users_emails_input.valid_email"
        value={Array []}
      />
    </div>
    <div
      className="help-text"
    >
      <MemoizedFormattedMessage
        defaultMessage="Add existing guests or send email invites to new guests."
        id="invitation_modal.guests.add_people.description"
      />
    </div>
  </div>
  <div
    className="add-channels"
    data-testid="channelPlaceholder"
  >
    <h5>
      <MemoizedFormattedMessage
        defaultMessage="Search and Add Channels"
        id="invitation_modal.guests.add_channels.title"
      />
    </h5>
    <div>
      <MemoizedFormattedMessage
        defaultMessage="Search and add channels"
        id="invitation_modal.guests.add_channels.placeholder"
      >
        <Component />
      </MemoizedFormattedMessage>
    </div>
    <div
      className="help-text"
    >
      <MemoizedFormattedMessage
        defaultMessage="Specify the channels the guests have access to."
        id="invitation_modal.guests.add-channels.description"
      />
    </div>
  </div>
  <div
    className="custom-message"
    data-testid="customMessage"
  >
    <a
      href="#"
      onClick={[Function]}
    >
      <MemoizedFormattedMessage
        defaultMessage="Set a custom message"
        id="invitation_modal.guests.custom-message.link"
      />
    </a>
    <div
      className="help-text"
    >
      <MemoizedFormattedMessage
        defaultMessage="Create a custom message to make your invite more personal."
        id="invitation_modal.guests.custom-message.description"
      />
    </div>
  </div>
  <div
    className="invite-guests"
  >
    <button
      className="btn btn-inactive"
      disabled={true}
      id="inviteGuestButton"
      onClick={[Function]}
    >
      <MemoizedFormattedMessage
        defaultMessage="Invite Guests"
        id="invitation_modal.guests.invite_button"
      />
    </button>
  </div>
</div>
`;
